tests/Annotations/Command/QueueWatchCommandTest.php

```diff
@@ -116,6 +119,63 @@ public function it_retries_failures()
         $this->assertSame(1, $this->queue->count());
     }
 
+    /**
+     * @test
+     */
+    public function it_will_remove_queue_item_if_upsert_fails()
+    {
+        $item = new InternalSqsMessage('profile', 'username');
+        $profile = new Profile(
+            'username',
+            new PersonDetails('PreferredName', 'IndexName'),
+            new EmptySequence(),
+            new EmptySequence()
+        );
+        $data = [
+            'authority' => $this->authority,
+            'username' => 'username',
+            'email' => 'username@blackhole.elifesciences.org',
+            'display_name' => 'PreferredName',
+        ];
+        $this->transformer
+            ->expects($this->once())
+            ->method('transform')
+            ->will($this->returnValue($profile));
+        $post_request = new Request(
+            'POST',
+            'users',
+            ['Authorization' => $this->authorization, 'User-Agent' => 'HypothesisClient'],
+            json_encode($data)
+        );
+        $post_response_mess = json_encode(['status' => 'failure', 'reason' => 'user with username username already exists.']);
+        $post_response = new Response(400, [], $post_response_mess);
+        $rejected_post_response = new RejectedPromise(new BadResponse($post_response_mess, $post_request, $post_response));
+        $this->httpClient
+            ->expects($this->at(0))
+            ->method('send')
+            ->willReturn($rejected_post_response);
+        $patch_data = [
+            'email' => 'username@blackhole.elifesciences.org',
+            'display_name' => 'PreferredName',
+        ];
+        $patch_request = new Request(
```

Comment:
I understand the fidelity to what happens in reality, but the `Request` objects could easily be dummies (containing for example just the `PATCH`|`POST` value) in this test since we don't read them.